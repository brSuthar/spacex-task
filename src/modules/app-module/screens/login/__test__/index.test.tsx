import 'react-native';
import React from 'react';
import Login from '..';

import {render, fireEvent} from '@testing-library/react-native';
import {it, expect, describe} from '@jest/globals';
import renderer from 'react-test-renderer';

describe(Login, function () {
  it('Should render user input', function () {
    const {getByTestId} = render(<Login />);
    const input = getByTestId('user-name');
    expect(input).toBeTruthy();
    expect(input.props.value).toBe('');
    expect(input.props.maxLength).toBe(10);
    expect(input.props.placeholder).toBe('Username');
    fireEvent.changeText(input, 'bhomaram suthar');
    expect(input.props.value).toBe('bhomaram suthar');
  });
  it('Should render password', function () {
    const {getByTestId} = render(<Login />);
    const password = getByTestId('password');
    expect(password).toBeTruthy();
  });
  it('Should click login', function () {
    const {getByTestId} = render(<Login />);
    fireEvent.press(getByTestId('login'));
  });

  describe('Render login', function () {
    it('login rendered correctly', function () {
      const tree = renderer.create(<Login />);
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });
});
