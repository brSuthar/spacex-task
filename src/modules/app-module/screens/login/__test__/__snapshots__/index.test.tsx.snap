// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Render login login rendered correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFF",
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 24,
    }
  }
>
  <TextInput
    maxLength={10}
    onChangeText={[Function]}
    placeholder="Username"
    style={
      {
        "borderColor": "gray",
        "borderRadius": 6,
        "borderWidth": 1,
        "height": 52,
        "marginBottom": 16,
        "paddingHorizontal": 16,
        "textAlignVertical": "center",
        "width": "100%",
      }
    }
    testID="user-name"
    value=""
  />
  <TextInput
    onChangeText={[Function]}
    placeholder="Password"
    secureTextEntry={true}
    style={
      {
        "borderColor": "gray",
        "borderRadius": 6,
        "borderWidth": 1,
        "height": 52,
        "marginBottom": 16,
        "paddingHorizontal": 16,
        "textAlignVertical": "center",
        "width": "100%",
      }
    }
    testID="password"
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#0000FF",
        "borderRadius": 6,
        "height": 52,
        "justifyContent": "center",
        "opacity": 1,
        "width": "80%",
      }
    }
    testID="login"
  >
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontSize": 16,
          "fontWeight": "700",
        }
      }
    >
      Login
    </Text>
  </View>
</View>
`;
